### ISSUE TITLE
setcolorder could accept a vector of column names of length lower than the number of columns

### ISSUE THREAD

**Issue by matthieugomez** [Rdatatable/data.table#798]
It would be nice if `setcolorder` could accept argument of length lower than the width of the datatable, so that

``` R
setcolorder(DT, "v3")
```

could by typed instead of

``` R
setcolorder( DT, c("v3", names(DT[names(DT) != "v3"])) )
```

As an outsider point of view, it is also more consistent with the fact that `setnames` can be used with one name instead of the whole list of names.

**Comment by MichaelChirico** [comment]
In fact my PR #1792 fixes #592 by implementing this issue. Just noticed this...

**Comment by mattdowle** [comment]
Yes, solved too thanks to @MichaelChirico 's PR #1762 now merged in v1.10.5.
